add-on/src/lib/on-installed.js

```diff
@@ -21,6 +21,15 @@ export async function onInstalled (details) {
 export async function runPendingOnInstallTasks () {
   const { onInstallTasks, displayReleaseNotes } = await browser.storage.local.get(['onInstallTasks', 'displayReleaseNotes'])
   await browser.storage.local.remove('onInstallTasks')
+  // this is needed because `permissions.request` cannot be called from a script. If that happens the browser will
+  // throws: Error: permissions.request may only be called from a user input handler
+  // To avoid this, we open a new tab with the permissions page and ask the user to grant the permissions.
+  // That makes the request valid and allows us to gain access to the permissions.
+  if (!(await browser.permissions.contains({ origins: ['<all_urls>'] }))) {
```

Comment:
if a user doesn't grant permissions in firefox, will they be constantly bombarded with the permissions page?